### PR TITLE
Add release notes for 6.1.3

### PR DESCRIPTION
**Pull request by hsiaoching**

### CODE REVIEW

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+=============================================================
+
+Summary
+-------
+
+This release introduces new features, performance improvements as well as bug fixes. Some of the highlights are:
+
+1. **Performance improvements**
+    - Improve the performance of joiner plugins to better handle data skewness with capped memory.
+    - Support a new aggregator API to achieve better performance for spark engine.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: s/spark/Spark/

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16816` - Fixed schedule properties to overwrite preferences set on the application instead of the other way around. This most visibly fixed a bug where the compute profile set on a pipeline schedule or trigger would get overwritten by the profile for the pipeline.
+
+- :cask-issue:`CDAP-16826` - Fixed an issue to not show page level error message when the backend is down.
+
+- :cask-issue:`CDAP-16845` - Fixed a bug that started running preview for pipelines with post-run actions even if user chose option to not run preview.
+
+- :cask-issue:`CDAP-16870` - Fixed PySpark support to work with Spark 2.1.3+.
+
+- :cask-issue:`CDAP-16876` - Removed leftover header for provided arguments in preview.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: remove

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-15775` - Fixed a bug that disallowed writing to an empty BigQuery table without any data or schema.
+
+- :cask-issue:`CDAP-16055` - Fixed a bug that the failure error message emitted by Spark driver is not being collected.
+
+- :cask-issue:`CDAP-16211` - Unified JSON structure used by REST endpoints for getting pipeline configuration and deploying pipelines.
+
+- :cask-issue:`CDAP-16222` - Fixed the package references in the dynamic spark plugin to use io.cdap instead of co.cask.
+
+- :cask-issue:`CDAP-16240` - Fixed a bug to show master and worker memory in dataproc compute profiles in GB.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Dataproc

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 4]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16725` - Fixed a bug where concurrent preview runs were failing because SparkConf for the new preview runs was getting populated with the configurations from the previously started in-progress preview run.
+
+- :cask-issue:`CDAP-16731` - Fixed a bug that the groupBy aggregator requires a different alias for the field name.
+
+- :cask-issue:`CDAP-16736` - Fixed record schema comparison to include record name.
+
+- :cask-issue:`CDAP-16751` - Fixed a bug where UI overwrites scale property of a decimal schema field if 0.
+
+- :cask-issue:`CDAP-16760` - Fixed a bug where memory, cpu, and engine config properties were not being set for sparkprogram plugins.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: CPU

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 5]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16788` - Remove reference to detailed view of an application. UI now only shows overview of custom applications and pipeline detailed view for pipelines when navigating from control center.
+
+- :cask-issue:`CDAP-16801` - Fixed an issue where runtime arguments would lose focus after typing certain properties.
+
+- :cask-issue:`CDAP-16809` - Added support for compressed file with header copying for text file based source.
+
+- :cask-issue:`CDAP-16816` - Fixed schedule properties to overwrite preferences set on the application instead of the other way around. This most visibly fixed a bug where the compute profile set on a pipeline schedule or trigger would get overwritten by the profile for the pipeline.
+
+- :cask-issue:`CDAP-16826` - Fixed an issue to not show page level error message when the backend is down.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: remove

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 6]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16975` - UI now adds the latest version of plugin, among the list of different versions of the plugin, when added from the sidepanel in pipeline studio. If the user has already chosen a specific version (older version) it defaults to that instead of the latest.
+
+- :cask-issue:`CDAP-16976` - UI resets the default version of plugins for specific user during upgrade. When users upgrade from 6.1.2 to 6.1.3 or later UI will reset the default version of plugin the user has already chosen. Post upgrade if the user uses the same plugin UI will choose the latest version of the same plugin.
+
+- :cask-issue:`CDAP-16993` - Fixed a bug in preview for fields that have non-string types such as bytes.
+
+- :cask-issue:`CDAP-17000` - Changed default value of spark.network.timeout to 10 minutes to make pipeline execution more stable for shuffle heavy pipelines.
+
+Improvements
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: movee improvements before bug fixes

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 7]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16621` - Removed modal showing pipeline JSON when users export pipelines. Instead, pipeline gets downloaded when users click "export pipeline" without the extra confirmation step.
+
+- :cask-issue:`CDAP-16656` - Added support for rendering large schemas (>1000 fields) in pipelines UI. By default collapse complex schemas and lazy-load fields in record types.
+
+- :cask-issue:`CDAP-16673` - Added payload compression support to messaging service.
+
+- :cask-issue:`CDAP-16676` - Upgrade to use dataproc API v1beta2 to allow endpoint config.
+
+- :cask-issue:`CDAP-16708` - Added a new AutoJoiner API for plugins to implement. The new API leaves implementation details up to the application, which can perform join optimizations that were not possible with the older Joiner API.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: this should be a new feature, not an improvement.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 8]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16711` - Added the ability for Joiner plugins to specify whether null keys should match other null keys.
+
+- :cask-issue:`CDAP-16815` - Added a metric records.updated in BigQuery sink. This will give a total of all the inserts, updates and upserts into the sink.
+
+- :cask-issue:`CDAP-16875` - Added two new properties to the Joiner plugin -- 'Join on Null Values' and 'Inputs to Load in Memory'. The plugin also uses new join capabilities that make it more resilient to running out of memory.
+
+- :cask-issue:`CDAP-16918` - Introduced a new REST API for getting all application details across all namespaces.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: new feature

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 9]
```diff
@@ -30,6 +30,158 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16929` - Added the ability to select a Custom Dataproc Image. The complete URI for the custom image should be specified.
+
+- :cask-issue:`CDAP-17003` - Removed redundant validations from BQ sink, this should reduce calls to BigQuery.getTable().
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: remove?

**Review by bdmogal — CHANGES_REQUESTED** [review]
some minor comments

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 3 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 4 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 5 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 6 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 7 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 8 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 9 above)
Comment (hsiaoching) [reply cdap-docs/reference-manual/source/release-notes.rst]: Done

**Review by bdmogal — APPROVED** [review]

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 10]
```diff
@@ -30,6 +30,156 @@ Cask Data Application Platform Release Notes
...
+1. **Performance improvements**
+    - Improve the performance of joiner plugins to better handle data skewness with capped memory.
+    - Support a new aggregator API to achieve better performance for Spark engine.
+
+2. **Platform and pipeline enhancements**
+    - Support upgrading all pipelines in a namespace via REST API to use latest available artifacts.
+    - Upgrade to use Dataproc API v1beta2 to allow endpoint configuration.
+    - Add various bug fixes and enhancements for preview and pipeline execution.
+    - Allow user to select a Custom Dataproc Image by specifying image URI.
+
```
Comment (nitinmotgi) [inline cdap-docs/reference-manual/source/release-notes.rst:55]: We should talk about upgrade.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 11]
```diff
@@ -30,6 +30,156 @@ Cask Data Application Platform Release Notes
...
+New Features
+------------
+
+- :cask-issue:`CDAP-16708` - Added a new AutoJoiner API for plugins to implement. The new API leaves implementation details up to the application, which can perform join optimizations that were not possible with the older Joiner API.
```
Comment (nitinmotgi) [inline cdap-docs/reference-manual/source/release-notes.rst]: Group all joiners together.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 12]
```diff
@@ -30,6 +30,156 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16709` - Implemented performance improvements to joiner plugins to cap the required memory to around 4gb per executor instead of scaling up as the skewness of the join goes up. Joins can now also be performed in-memory if one side is small, and behavior on null keys can be chosen by the user.
+
+- :cask-issue:`CDAP-16835` - Added support for upgrading applications via REST API. Example usage is to upgrade all pipelines in a namespace to use latest available artifacts.
```
Comment (nitinmotgi) [inline cdap-docs/reference-manual/source/release-notes.rst]: All upgrades together.

**Pull request closed without merging**